### PR TITLE
Work on CheckForTraktTokenUpdate

### PR DESCRIPTION
**Pull request by krbrs**
Have it read the expiration time from settings and not just refresh the token every 24h, only if it's going to expire in 45 days

### CODE REVIEW

**Comment by krbrs** [comment]
I could not test expiration and renewal yet. But at least I get a proper output of the token valid date in the logs while it's still valid.

If someone can look into this code and optimize it if needed (or give feedback), that would be great

File: Shoko.Server/Import/Importer.cs [hunk 1]
```diff
@@ -1438,49 +1438,41 @@ public static void CheckForTraktTokenUpdate(bool forceRefresh)
...
-            var freqHours = 24; // we need to update this daily
 
-            var sched =
-                RepoFactory.ScheduledUpdate.GetByUpdateType((int)ScheduledUpdateType.TraktToken);
-            if (sched != null)
+            // Convert the Unix timestamp to DateTime directly
+            var expirationDate = DateTimeOffset.FromUnixTimeSeconds(long.Parse(settings.TraktTv.TokenExpirationDate)).DateTime;
+
+            // Check if force refresh is requested or the token is expired
+            if (forceRefresh || DateTime.Now.Add(TimeSpan.FromDays(45)) >= expirationDate)
```
Comment (da3dsoul) [inline Shoko.Server/Import/Importer.cs:1452]: Why 45 days?

Comment (krbrs) [reply Shoko.Server/Import/Importer.cs:1452]: https://discord.com/channels/96234011612958720/1148092356592160768/1195127627397214331

Initially I had it set to 30 days, but half would be 45 days so I set this. We can set it to whatever is fine

Comment (da3dsoul) [reply Shoko.Server/Import/Importer.cs:1452]: Half of what

Comment (da3dsoul) [reply Shoko.Server/Import/Importer.cs:1452]: Fair enough. Looks good then

**Pull request merged**